backend/cfg/vectorize.ml

```diff
@@ -2,13 +2,180 @@
 
 module DLL = Flambda_backend_utils.Doubly_linked_list
 
-let dump ppf cfg_with_layout ~msg =
+let (<<) f g x = f (g x)
+
+module Instruction : sig
+  (* CR-someday tip: consider moving this to cfg or at least have something similar there *)
+  module Id : sig
+    type t
+
+    include Identifiable.S with type t := t
+
+    val to_int : t -> int
+  end
+
+  type t =
+    | Basic of Cfg.basic Cfg.instruction
+    | Terminator of Cfg.terminator Cfg.instruction
+
+  val id : t -> Id.t
+
+  val args : t -> Reg.t Array.t
+
+  val ress : t -> Reg.t Array.t
```

Comment:
I find `ress` confusing. I prefer `arg` and `res` to match `Cfg.instruction` fields. Their types indicate that it's not a single argument / result register   .